tensorflow/compiler/mlir/tensorflow/transforms/lower_tf.td

```diff
@@ -457,3 +457,45 @@ def LowerIsFiniteOp : Pat<(TF_IsFiniteOp $x),
                           (TF_SubOp $x, $x),
                           (TF_ConstOp (GetScalarOfType<0> $x)),
                           /*incompatible_shape_error*/ConstBoolAttrTrue)>;
+
+//===----------------------------------------------------------------------===//
+// Selu op patterns.
+//===----------------------------------------------------------------------===//
+
+def : Pat<(TF_SeluOp: $src_op AnyRankedTensor:$features),
+        (TF_SelectOp
+          (TF_GreaterEqualOp
+            $features,
+            (TF_FillOp
+              (CreateTFShapeOp $src_op, $features, /*use 32bit*/ConstBoolAttrFalse),
+              (TF_ConstOp (GetScalarOfType<0> $features)))),
+          (TF_MulOp
+            $features,
+            (TF_FillOp
+              (CreateTFShapeOp $src_op, $features, /*use 32bit*/ConstBoolAttrFalse),
+              (TF_ConstOp (GetScalarOfFloatType<"1.0507009873554804934193349852946"> $features)))),
+          (TF_MulOp
+            (TF_Expm1Op $features),
+            (TF_FillOp
+              (CreateTFShapeOp $src_op, $features, /*use 32bit*/ConstBoolAttrFalse),
+              (TF_ConstOp (GetScalarOfFloatType<"1.7580993408473768599402175208123"> $features)))))>;
+
+def : Pat<(TF_SeluGradOp:$src_op AnyStaticShapeTensor:$gradients, AnyRankedTensor:$features),
+        (TF_SelectOp
+          (TF_GreaterEqualOp
+            $features,
+            (TF_FillOp
+              (CreateTFShapeOp $src_op, $features, /*use 32bit*/ConstBoolAttrFalse),
+              (TF_ConstOp (GetScalarOfType<0> $features)))),
+          (TF_MulOp
+            $gradients,
+            (TF_FillOp
+              (CreateTFShapeOp $src_op, $features, /*use 32bit*/ConstBoolAttrFalse),
+              (TF_ConstOp (GetScalarOfFloatType<"1.0507009873554804934193349852946"> $features)))),
```

Comment:
define a const helper for these so that the same constant doesn't need to be defined twice. We might be able to reuse GetScalarOfFloatType for this.

Reply:
Done. Please re-review @smit-hinsu